provider/gcp/sia-gce/authn.go

```diff
@@ -44,16 +44,24 @@ func GetGCEConfig(configFile, profileConfigFile, metaEndpoint, region string, pr
 		}
 	}
 	if profileConfig == nil {
-		profileConfig, err := GetGCEAccessProfile(profileConfigFile, metaEndpoint, provider)
+		profileConfig, err := GetGCEAccessProfile(profileConfigFile, profileRestrictToKey, metaEndpoint, provider)
 		if err != nil {
 			log.Printf("Unable to determine user access management profile information: %v\n", err)
 		}
 		return config, profileConfig, nil
 	}
+	if profileConfig.ProfileRestrictTo == "" && profileRestrictToKey != "" {
```

Comment:
We're repeating this block in two places. So we can leave this logic in this method and remove it from the GetGCEAccessProfile method. Then we just have to remove line 51 where we return from this method and then add profileConfig != nil to the if line on line 53. This way we only have the logic in a single place and not multiple.